### ISSUE TITLE
[feature] tags

### ISSUE THREAD

**Issue by jrpie** [jrpie/Launcher#84]
(requested by Sarp Küçük)
#  Please describe the problem to be solved

Some apps are not important enough to show them in all apps but it is also tedious to unhide them whenever they are needed.

#  Describe the solution you would suggest

Let users add tags to apps. For each tag:
* add an action to show a list of apps with this tag
* add an option to hide apps with this tag from all apps
* add an option to show this tag in the drawer (and also respect it when searching) (?)

Furthermore add an action to list all tags.

#  Describe alternative solutions you've considered

Make hidden apps more easily accessibile. However the main reason to add the hidden apps feature was that users wanted to hide preinstalled junk apps they __never__ use. So this is not an option.

#  Additional info

This might unify the features "hidden apps" and "favorite apps", as hidden and favorite could be implemented as tags with special handling.

**Comment by jrpie** [comment]
I don't really like the idea.
The main reason to implement hidden apps was that users wanted to hide unremovable bloatware (i.e. junk which is never used). The idea of the app list is to find __every__ app quickly, so I don't see why it would be desirable to hide apps that are infrequently used. I'm not even sure whether implementing favorite apps was a good idea; personally I don't use that feature.
A tag-based launcher is simply a different approach.
Imho implementing two competing approaches is not a good idea.

**Comment by milesmcclane** [comment]
> > Sometimes I forget the name of an app, but remember it once I see it.
> 
> For me renaming helps a lot. For example, I tend to forget the name [Etar](https://github.com/Etar-Group/Etar-Calendar), but renaming it to "Calendar | Etar" helps a lot.
> 
> On the other hand, I think tags could be implemented without adding too much complexity and without getting in the way.

I agree. I rename things now myself. But it does become problematic on occasion when I need to know the real name for troubleshooting or searching something regarding the app. I admit - not major problems - but I’m glad to hear you’re considering tags. 

I moved from mlauncher due to some dubious additions (which felt like a detriment to my priority of privacy) which was a perfect minimalist launcher, to Kiss. Kiss is great but it lacks the minimalist ideals I crave. That’s why when I discovered your app I was so pleased. I do miss tags from kiss though, they weren’t there in mlauncher and I didn’t know what I was missing out on. It’s just a useful feature to have for organisation and use if you choose.

**Comment by jrpie** [comment]
Yes, that is the main argument against it imo

**Comment by jrpie** [comment]
On the other hand I understand silverplasma's use case (see chat). 
Selecting the tag "messenger" and then the one you want, might be preferable to typing (especially on a small screen), i.e. an efficiency gain, not a distraction.

**Comment by milesmcclane** [comment]
> On the other hand I understand silverplasma's use case (see chat). Selecting the tag "messenger" and then the one you want, might be preferable to typing (especially on a small screen), i.e. an efficiency gain, not a distraction.

Exactly. I want to type ‘running’, and then have all my related apps at hand which I use when going for a run. ‘Driving’ and have music and navigation apps appear. Etc

**Comment by jrpie** [comment]
> > On the other hand I understand silverplasma's use case (see chat). Selecting the tag "messenger" and then the one you want, might be preferable to typing (especially on a small screen), i.e. an efficiency gain, not a distraction.
> 
> Hello, I am silverplasma on chat. Please excuse the late reply.

Nothing to apologize for, we are not in a hurry in a hurry (I don't have too much time myself at the moment ^^)
> 
> My use case for tags is one of organization as per the users needs/use-cases. I am not advocating to have apps pre-tagged. The tagging system/feature would be enabled/used at the users own discretion.
> 
> I have a lot of apps on my phone. Some I use often while other I use only during specific scenarios. Some apps have easy names while other have some pretty unique names.
> 
> * I would like to be able to with ease have all the apps I use for a giving scenario available at a swipe, 2-3 typed letters or just a tap away.
> * Ideally this tag/filtered selection would stay on the slide all-apps slide menu till the user selects the ALL tag or clears the tag. Perhaps this can be feature one can turn on and  off.
> * Apps can be assigned more than one tag. For example I would have my Camera tagged with **PhoGrph** but also with **Nav** for when driving or hiking.
> * Ideally the presentation of the tagged apps can be manually organized. Such that regardless it he user adds a new app, the position of apps does not change. This is the issue Favorite apps have where when new apps are favorite, the organization of the Favorite menu is altered.
> 
> The idea is to be able to have access to these apps under a given tag(s) during the given scenario.
> 
> For example: I tagged all the apps I used for photography and image editing under **PhoGrph**. Another user could use **ImgEd** or **IE**. _Again, it is up to the user to tag their own apps_. I either swipe, type or select **PhoGrph** to have all other apps filtered out. I would then select my camera to take a picture. then I would again access the all-app slide menu -which remembers that I am using the **PhoGrph** tag, then select AvesLibres to check my picture, edit the meta data then from there share to one of my image editing apps. Then again navigate to the all-apps slide menu, which remember the tag I am using, and select Mixplorer to manage my files and organize my images locally and on my server.
> 
> Same type of mLaucher flow for my **Workout**, **Navigation** and **Tools** scenarios where I would like to have a selected group of apps readily available.
> 
> I produced and image inspired by one shared by josia on chat. Just an idea of what tags on the all-app slide menu could look like.

**Comment by jrpie** [comment]
> Hello, I am silverplasma on chat. Please excuse the late reply.

Nothing to apologize for, we are not in a hurry in a hurry (I don't have too much time myself at the moment ^^)

> My use case for tags is one of organization as per the users needs/use-cases. I am not advocating to have apps pre-tagged. The tagging system/feature would be enabled/used at the users own discretion.

Yes, I definitely won't pre-tag anything, there are way too many apps out there, and clearly users will have their own strategies of oganizing.

> Ideally this tag/filtered selection would stay on the slide all-apps slide menu till the user selects the ALL tag or clears the tag. Perhaps this can be feature one can turn on and off.

Not really sure how to do that. If the tags are hard to access, what use do they have? (I could perhaps add different actions for lists with / without the tag bar)

> Apps can be assigned more than one tag. For example I would have my Camera tagged with PhoGrph but also with Nav for when driving or hiking.

Yes, that is definitely planned. (Otherwise the feature would be called "folders" or "directories")

> Ideally the presentation of the tagged apps can be manually organized. [...]

Thats quite hard to do, requires a lot of UI and imo against the core ideas of µLauncher. What should be displayed if there is an ordering for tag A and one for tag B, but then the user filters apps that belong to A and B?
(To me this sounds more like "organizing stuff inside a folder" than "have tags that filter the list and make searching quicker", it is just not really compatible)

> I produced and image inspired by one shared by josia on chat. Just an idea of what tags on the all-app slide menu could look like.

Yes, that is pretty much what I had in mind 👍🏼

I plan that for every tag the user can decide whether or not to show it in the tag bar in the "all apps" list. Furthermore users will also be able to bind a custom action that directly shows a list of apps filtered by certain tags. That way very important tags could be just bound to a gesture.
